File: P010/main.java
import java.lang.*;

public class main {
    private static boolean isPrime(int n) {
        if(n == 1)
            return false;

        for(int i=2; i<=Math.sqrt(n); i++)
            if(n%i == 0)
                return false;
        return true;
    }

    private static void sumOfPrimes(int n) {
        long result = 0;
        for(int i=2; i<n; i++)
            if(isPrime(i))
                result += i;

        System.out.println(result);
    }

    public static void main(String[] args) {
        long startTime = System.currentTimeMillis();

        sumOfPrimes(2000000);

        long endTime = System.currentTimeMillis();
        long totalSpan = System.currentTimeMillis();
        System.out.println("Completed in " + totalSpan + " milliseconds.");
    }
}

File: P010/main.py
import time
import math

def isPrime(n) :
    if n == 1 : return False

    i = 2
    while i <= math.sqrt(n) :
        if n%i==0 :
            return False
        i+=1
    return True

def sumOfPrimes(n) :
    result = 0
    i = 2
    while i < n :
        if isPrime(i) == True :
            result += i
        i+=1

    print result

start = time.time()

sumOfPrimes(2000000)

elapsed = time.time()

print("found in %s seconds") % (elapsed)
